packages/util-node-http-modules/src/util-node-http-modules.js

```diff
@@ -1,12 +1,19 @@
 export async function getNodeHttpModules() {
-  let nodeHttpsTransport
-  try {
-    nodeHttpsTransport = await import("https").catch(e => null)
-  } catch (e) {}
-  let nodeHttpTransport
-  try {
-    nodeHttpTransport = await import("http").catch(e => null)
-  } catch (e) {}
+  let nodeHttpsTransport = null
+  let nodeHttpTransport = null
+
+  if (
+    typeof process !== "undefined" &&
+    Object.prototype.toString.call(process) === "[object process]"
+  ) {
+    // bypass webpack's lazy missing modules check using variables
+    // browser should never reach this scope anyways so webpack should not be checking whether these modules exist
+    const httpsModule = "https"
+    const httpModule = "http"
+
+    nodeHttpsTransport = await import(httpsModule).catch(e => null)
+    nodeHttpTransport = await import(httpModule).catch(e => null)
```

Comment:
This is necessary. When webpack bundles, it goes through all import statements and if any import node modules, it throws a ModuleNotFoundError - this even happens if the module import is in an unreachable scope.

This must be done to trick webpack.